### PR TITLE
core: add new CLS (all frames) to hidden metrics audit

### PR DESCRIPTION
**Pull request by brendankenny**
first checkbox of https://github.com/GoogleChrome/lighthouse/issues/12350

It should still be easy to copy and paste out of the variants file into the real metric file when the time comes. Doing it now in the `metrics` audit so the tail end of the May HTTP Archive run can pick up some CLS AF data with the new CLS definition.

### CODE REVIEW

File: types/artifacts.d.ts [hunk 1]
```diff
@@ -789,6 +789,7 @@ declare global {
         layoutShiftMaxSessionGap1sLimit5s: number,
         layoutShiftMaxSliding1s: number,
         layoutShiftMaxSliding300ms: number,
+        newCumulativeLayoutShiftAllFrames: number,
```
Comment (patrickhulce) [inline types/artifacts.d.ts:792]: I didn't expect `new` to be part of the internal name :)

`layoutShiftMaxSessionGap1sLimit5sAllFrames`? :D

**Review by patrickhulce — COMMENTED** [review]
impl LGTM

I have the open naming question mostly because I'm not sure where this PR fits in with the overall plan for new CLS. If it's a very temporary thing that will disappear by 8.0, then as-is WFM. If it's being engraved permanently in metrics.js today, then let's chat :)

**Comment by brendankenny** [comment]
> I have the open naming question mostly because I'm not sure where this PR fits in with the overall plan for new CLS. If it's a very temporary thing that will disappear by 8.0, then as-is WFM. If it's being engraved permanently in metrics.js today, then let's chat :)

ha, no, that's very much what I was planning, but I guess I didn't discuss with others more than here and in #12046. Basically live in `metrics.js` for collection today, but I was thinking `layout-shift-variants` (and its additions to `metrics.js`) wouldn't survive to 8.0, taking what works in it and in `layout-shift-variants-test` and moving it to the CLS, then deleting the rest.

File: lighthouse-core/computed/layout-shift-variants.js [hunk 2]
```diff
@@ -71,6 +71,40 @@ class LayoutShiftVariants {
...
+  /**
+   * Calculates cumulative layout shifts per cluster (session) of LayoutShift
+   * events -- where a new cluster is created when there's a gap of more than
+   * 1000ms since the last LayoutShift event or the cluster is greater than
+   * 5000ms long -- and returns the max LayoutShift score found.
+   * @param {Array<LayoutShiftEvent>} layoutShiftEvents
+   * @return {number}
+   */
+  static newCumulativeLayoutShift(layoutShiftEvents) {
+    const gapµs = 1_000_000;
```
[the remark this answers is not in the document]
Comment (brendankenny) [reply lighthouse-core/computed/layout-shift-variants.js:83]: > how sad would it be to use?

not sad at all :)

> (not to mention problematic with the fact that all of our camelcase millisecond `Ms` are actually indistinguishible from capital Mu `Ms`)

ah, see, there's the problem. You're still writing `M` when clearly you mean `Μ` :D

**Comment by patrickhulce** [comment]
> I was thinking layout-shift-variants (and its additions to metrics.js) wouldn't survive to 8.0, taking what works in it and in layout-shift-variants-test and moving it to the CLS, then deleting the rest.

Cool beans, that's what I figured, but wanted to double check just in case :)

**Review by patrickhulce — APPROVED** [review]

File: types/artifacts.d.ts (hunk 1 above)
Comment (brendankenny) [reply types/artifacts.d.ts:792]: `layoutShiftMaxSessionGap1sLimit5sAllFrames` is good

**Pull request merged**